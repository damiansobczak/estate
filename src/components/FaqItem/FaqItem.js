import React from "react";
import "./FaqItem.scss";

class FaqItem extends React.Component {
  render() {
    return (
      <div className="column is-6-tablet">
        <div className="faqitem">
            <div className="faqitem__count">{this.props.count}.</div>
            <div className="faqitem__content">
                <div className="faqitem__title">{this.props.title}</div>
                <div className="faqitem__desc">{this.props.desc}</div>
            </div>
        </div>
      </div>
    );
  }
}

export default FaqItem;
