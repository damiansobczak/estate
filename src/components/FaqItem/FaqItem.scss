@import "../../assets/styles/utils/variables";
@import "../../assets/styles/utils/mixins";

.faqitem {
    border-top: 1px solid $color-neutral-200;
    display: flex;
    padding: 40px 40px 0 0;
    margin-bottom: 50px;
    box-sizing: border-box;

    &__count {
        color: $color-accent;
        font-family: poppins, sans-serif;
        font-weight: 700;
        font-style: normal;
        margin-right: 40px;
    }

    &__title {
        font-family: poppins, sans-serif;
        font-weight: 700;
        font-style: normal;
        color: $color-text-primary;
        margin-bottom: 15px;
        line-height: 1.2;
    }

    &__desc {
        color: $color-text-secondary;
        font-family: proxima-nova, sans-serif;
        font-weight: 500;
        font-style: normal;
        line-height: 1.5;
    }
}