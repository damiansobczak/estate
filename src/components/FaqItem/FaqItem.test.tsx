import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import FaqItem from "./FaqItem";

describe("<FaqItem.test />", () => {
    it("renders component correctly", () => {
        const tree = renderer.create(<FaqItem count="01" title="To jest faq 1" desc="To jest desc dla Faq 01" />).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it("displays title correctly", () => {
        const wrapper = shallow(<FaqItem count="01" title="To jest faq 1" desc="To jest desc dla Faq 01" />);
        expect(wrapper.find(".faqitem__count").html()).toContain("01.");
    });

    it("displays title correctly", () => {
        const wrapper = shallow(<FaqItem count="01" title="To jest faq 1" desc="To jest desc dla Faq 01" />);
        expect(wrapper.find(".faqitem__title").html()).toContain("To jest faq 1");
    });

    it("displays description correctly", () => {
        const wrapper = shallow(<FaqItem count="01" title="To jest faq 1" desc="To jest desc dla Faq 01" />);
        expect(wrapper.find(".faqitem__desc").html()).toContain("To jest desc dla Faq 01");
    });
});
