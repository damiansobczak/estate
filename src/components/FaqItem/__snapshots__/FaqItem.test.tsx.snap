// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FaqItem.test /> renders component correctly 1`] = `
<div
  className="column is-6-tablet"
>
  <div
    className="faqitem"
  >
    <div
      className="faqitem__count"
    >
      01
      .
    </div>
    <div
      className="faqitem__content"
    >
      <div
        className="faqitem__title"
      >
        To jest faq 1
      </div>
      <div
        className="faqitem__desc"
      >
        To jest desc dla Faq 01
      </div>
    </div>
  </div>
</div>
`;
